File: server/QCLab/Client/Services/AuthService.cs
using System.Net.Http.Json;
using QCLab.Client.Dtos;
using QCLab.Client.Services.Interfaces;

namespace QCLab.Client.Services;

public class AuthService : IAuthService
{
    private readonly HttpClient _httpClient;

    public AuthService(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

    public async Task<UserSessionDto?> Login(LoginRequest request)
    {
        var response = await _httpClient.PostAsJsonAsync("api/auth/login", request);
        if (response.IsSuccessStatusCode)
        {
            return await response.Content.ReadFromJsonAsync<UserSessionDto>();
        }
        return null;
    }

    public async Task Logout()
    {
        await _httpClient.PostAsync("api/auth/logout", null);
    }

    public async Task<UserSessionDto?> GetCurrentUser()
    {
        try
        {
            return await _httpClient.GetFromJsonAsync<UserSessionDto>("api/auth/me");
        }
        catch
        {
            return null;
        }
    }

    public async Task<bool> ChangePassword(ChangePasswordDto dto)
    {
        var response = await _httpClient.PostAsJsonAsync("api/auth/change-password", dto);
        return response.IsSuccessStatusCode;
    }
}

File: packages/client_react/src/services/authService.ts
import apiClient from './apiClient';
import type { ChangePasswordDto, LoginRequest, UserSessionDto } from '../types/auth';

const authService = {
  async login(request: LoginRequest): Promise<UserSessionDto> {
    const response = await apiClient.post<UserSessionDto>('api/auth/login', request);
    return response.data;
  },

  async logout(): Promise<void> {
    await apiClient.post('api/auth/logout');
  },

  async me(): Promise<UserSessionDto | null> {
    try {
      const response = await apiClient.get<UserSessionDto>('api/auth/me');
      if (response.status === 204 || !response.data) return null;
      return response.data;
    } catch (error) {
      return null;
    }
  },

  async changePassword(dto: ChangePasswordDto): Promise<void> {
    await apiClient.post('api/auth/change-password', dto);
  },
};

export default authService;
